reservation_units/models.py

```diff
@@ -573,22 +572,10 @@ def hauki_resource_data_source_id(self):
         return settings.HAUKI_ORIGIN_ID
 
     @property
-    def state(self) -> str:
-        now = datetime.datetime.now(tz=get_default_timezone())
-        if self.is_archived:
-            return ReservationUnitState.ARCHIVED
-        elif self.is_draft:
-            return ReservationUnitState.DRAFT
-        elif (self.publish_begins and now < self.publish_begins) or (
-            self.publish_ends and now >= self.publish_ends
-        ):
-            return ReservationUnitState.SCHEDULED_PUBLISHING
-        elif (self.reservation_begins and now < self.reservation_begins) or (
-            self.reservation_ends and now >= self.reservation_ends
-        ):
-            return ReservationUnitState.SCHEDULED_RESERVATION
-        else:
-            return ReservationUnitState.PUBLISHED
+    def state(self) -> ReservationUnitState:
+        from reservation_units.utils.reservation_unit_state_helper import get_state
+
+        return get_state(self)
```

Comment:
this can raise a `ValueErrror` should it be handled?

Reply:
The problem is that currently it is not possible to get this exception. `is_published` is basically an `else` branch but I didn't want to call it such because it is more complex in `get_is_published_query()`. So I don't really know how this could be tested 🤔 

I added the `throw` there for the future cases where we end up changing the definitions and somehow forget to cover some corner case. In that case it will throw an exception instead of just swallowing the missing case with `else`.

What are your thoughts on this? 😄

Reply:
Well this is getting kind of a test coverage issue then, but now we have code block that isn't tested. I like the idea that it will throw an error in future if those states are edited / add so removing it is something that we don't want.
Some test for a new type of enum value could be therefore invented if possible.

Reply:
Actually, now that I think of it, my reasoning is not very solid for two reasons:

1) We are using enums. If we ever add a new value, it is very easy to find the code where enum is used and update it
2) We are expecting tests for all enum values. If we ever add a new value, we need to write a test for it. That requires that we also update the implementation.

As a bonus reason, my idea hugs the magical premature optimization territory. I'm doing stuff based on my own expectations of the future. Not gud!

So I think the idea of raising "safety exception" is kinda pointless because the style the code is implemented and our testing requirements already handle everything.

I'll change the implementation to use if-else statement. It'll get rid of this nasty untested code issue ☺️